Mage/Logger/Writers/ConsoleWriter.cs

```diff
@@ -1,157 +1,201 @@
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Info(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.Log(makeLogString("info", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.Log(logEntry.data);
+		private void Info(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.Log(makeLogString("info", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Notice(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.Log(makeLogString("notice", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.Log(logEntry.data);
+		private void Notice(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.Log(makeLogString("notice", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Warning(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogWarning(makeLogString("warning", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.LogWarning(logEntry.data);
+		private void Warning(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogWarning(makeLogString("warning", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.LogWarning(logEntry.Data);
 			}
 		}
 
-		private void Error(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogError(makeLogString("error", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.LogError(logEntry.data);
+		private void Error(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogError(makeLogString("error", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.LogError(logEntry.Data);
 			}
 		}
 
-		private void Critical(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogError(makeLogString("critical", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				if (logEntry.data is Exception && (logEntry.data as Exception).StackTrace != null) {
-					Exception excpt = logEntry.data as Exception;
-					UnityEngine.Debug.LogError(excpt.ToString() + ":\n" + excpt.StackTrace.ToString());
-				} else {
-					UnityEngine.Debug.LogError(logEntry.data);
+		private void Critical(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogError(makeLogString("critical", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				if (logEntry.Data is Exception && (logEntry.Data as Exception).StackTrace != null)
+				{
+					var excpt = logEntry.Data as Exception;
+					UnityEngine.Debug.LogError(excpt + ":\n" + excpt.StackTrace);
+				}
+				else
+				{
+					UnityEngine.Debug.LogError(logEntry.Data);
 				}
 			}
 		}
```

Comment:
This function seems over-complicated for nothing ?
We can do the same without "is", without 2 level of indentation and multiple cast
```
// log entry wall
UnityEngine.Debug.LogError(makeLogString("critical", logEntry.Context, logEntry.Message));

// null check data
if (logEntry.Data == null)
{
    return;
}

// for exception
var exception = logEntry.Data as Exception;
if (exception != null && exception.StackTrace != null)
{
    UnityEngine.Debug.LogError(exception + ":\n" + exception .StackTrace);
    return;
}

// for other data type
UnityEngine.Debug.LogError(logEntry.Data);
```